lib/nest_thermostat/nest.rb

```diff
@@ -144,24 +140,19 @@ def temperature_scale=(scale)
     alias_method :temp_scale=, :temperature_scale=
 
     def fan_mode
-      status["device"][device_id]["fan_mode"]
+      device["fan_mode"]
     end
 
     def fan_mode=(state)
-      HTTParty.post(
-        "#{self.transport_url}/v2/put/device.#{self.device_id}",
-        body: %Q({"fan_mode":"#{state}"}),
-        headers: self.headers
-      ) rescue nil
+      post_to_device_api(fan_mode: state)
     end
 
     def method_missing(name, *args, &block)
-      if %i[away leaf].include?(name)
-        warn "`#{name}' has been replaced with `#{name}?'. Support for " +
-             "`#{name}' without the '?' will be dropped in future versions."
-        return self.send("#{name}?", *args)
-      end
-
+        if %i[away leaf].include?(name)
+           warn "`#{name}' has been replaced with `#{name}?'. Support for " +
```

Comment:
Use 2 (not 3) spaces for indentation.<br>Use `\` instead of `+` or `<<` to concatenate those strings.